models/repo/repo_list.go

```diff
@@ -735,3 +735,31 @@ func GetUserRepositories(opts *SearchRepoOptions) (RepositoryList, int64, error)
 	repos := make(RepositoryList, 0, opts.PageSize)
 	return repos, count, db.SetSessionPagination(sess, opts).Find(&repos)
 }
+
+func GetPrimaryRepoLanguageList(ctx context.Context, repos RepositoryList) (LanguageStatList, error) {
+	languageList := make(LanguageStatList, 0)
+
+	q := db.GetEngine(ctx).
```

Comment:
I don't think this will perform well in large instances. It would be better to get this data form FTS

Reply:
Discussed at https://github.com/go-gitea/gitea/pull/24212#discussion_r1172249016

Unsure what the ideal alternative would be

Reply:
Maybe add a new table to store the languages and counts

Reply:
> Maybe add a new table to store the languages and counts

Hmm, might work for the global list but unsure how to  filter them by the repo owner since a few databases do not support storing primitive types as an array. Might be possible to create yet another table containing a the list of owners but tbf I wonder if the benefits would be worth it at that point